src/Persistent_cohomology/include/gudhi/Persistent_cohomology.h

```diff
@@ -208,6 +208,44 @@ class Persistent_cohomology {
     }
   }
 
+  /**
+   * @private
+   * Temporary patch to make the computation of more general FilteredComplex possible
+   * Will be removed when a better solution was agreed on.
+   */
+  void compute_persistent_cohomology_without_optimizations(Filtration_value min_interval_length = 0) {
```

Comment:
The modification is small, well localized and doesn't seem to make a difference in the output, nor in the timings.
I think it would be safer to not repeat ourself here:
```cpp
  template <bool Optimization = true>
  void compute_persistent_cohomology(Filtration_value min_interval_length = 0) {
  ...
        if (Optimization) {
          switch (dim_simplex) {
          ...
          }
        } else {
          update_cohomology_groups(sh, dim_simplex);
        }
  ...
  }
```
Or

```cpp
  void compute_persistent_cohomology(Filtration_value min_interval_length = 0, bool optimization = true) {
```